frontend/src/__tests__/integration/hooks/TestHook.tsx

```diff
@@ -0,0 +1,227 @@
+export const TestHook = <T extends any[]>({
+  hook,
+  defaultHookParams,
+  hookParams = [],
+}: TestHookProps<T>) => {
+  const [selectedParam, setSelectedParam] = useState<T>(defaultHookParams);
+
+  // call hook
+  const hookResult = hook(...selectedParam);
+
+  // convert unknown hook response to unknown[]
+  const hookResultArray: unknown[] = [];
+  if (typeof hookResult === 'object') {
+    hookResultArray.push(...Object.values(hookResult ?? {}));
+  } else {
+    hookResultArray.push(hookResult);
+  }
+
+  const previousPropsRef = useRef(defaultHookParams);
+  const propStatusesRef = useRef<string[]>([]);
+
+  const previousResultArrayRef = useRef(hookResultArray);
+  const resultArrayStatusesRef = useRef<string[]>([]);
+
+  const previousResultRef = useRef(hookResult);
+  const resultStatusRef = useRef<string>('');
+
+  // get prop stability
+  defaultHookParams.forEach((prop, index) => {
+    const isStable = previousPropsRef.current[index] === prop;
+    propStatusesRef.current[index] = isStable ? 'stable' : 'unstable';
+  });
+  previousPropsRef.current = defaultHookParams;
+
+  // get whole result stability
+  const isStable = previousResultRef.current === hookResult;
+  resultStatusRef.current = isStable ? 'stable' : 'unstable';
+  previousResultRef.current = hookResult;
+
+  // get individual result stability
+  hookResultArray.forEach((prop, index) => {
+    const isStable = previousResultArrayRef.current[index] === prop;
+    resultArrayStatusesRef.current[index] = isStable ? 'stable' : 'unstable';
+  });
+  previousResultArrayRef.current = hookResultArray;
+
+  const getType = (value: unknown) => {
+    if (typeof value === 'object') {
+      if (Array.isArray(value)) {
+        return 'array';
+      }
+      return 'object';
+    }
+    return typeof value;
+  };
+
+  const isPrimitive = (value: unknown) => value !== Object(value);
+
+  const renderTable = (
+    statusRef: string[],
+    dataArray: unknown[],
+    dataIdPrefix: string,
+    showButtons = false,
+  ) => (
+    <TableComposable variant="compact">
+      <Thead>
+        <Tr>
+          <Th>Type</Th>
+          <Th>Value</Th>
+          <Th>Status</Th>
+          <Th />
+        </Tr>
+      </Thead>
+      <Tbody>
+        {dataArray.map((value, index) => (
+          <Tr key={`table-row-${dataIdPrefix}-${index}`}>
+            <Td dataLabel="type" data-testid={`${dataIdPrefix}-type-${index}`}>
+              {getType(value)}
+            </Td>
+            <Td width={70} dataLabel="value" data-testid={`${dataIdPrefix}-value-${index}`}>
+              {typeof value === 'function' ? String(value) : JSON.stringify(value)}
+            </Td>
+            <Td dataLabel="status" data-testid={`${dataIdPrefix}-status-${index}`}>
+              <span style={{ color: statusRef[index] === 'stable' ? 'green' : 'red' }}>
+                {!isPrimitive(value) && statusRef[index]}
+              </span>
+            </Td>
+            <Td isActionCell>
```

Comment:
I'm sorry cause I can't wrap my head around this, why are we adding this button?

Reply:
[UPDATE] We had an internal discussion and @Gkrumbach07 answered all my questions.